tensorflow/python/eager/forwardprop.py

```diff
@@ -140,6 +140,48 @@ def _jvp_helper(op_name, attr_tuple, inputs, outputs, tangents):
     return output_tangents
 
 
+def _jvp_helper_wrapper(
+    op_name, attr_tuple, inputs, outputs, tangents, batch_size
+):
+    """Computes a batch of Jacobian-vector product for an op.
+
+  Args:
+    op_name: A string, the type of operation being executed.
+    attr_tuple: Attributes of the operation.
+    inputs: A flat list of input Tensors to the operation.
+    outputs: A flat list of output Tensors from the operation.
+    tangents: A flat list of Tensors, same shape as `[batch_size] + input_shape`.
+
+  Returns:
+    A flat list of tangents corresponding to `outputs`.
+  """
+  use_pfor = False
+  if batch_size is not None:
+    use_pfor = True
```

Comment:
Do we need `use_pfor`? It seems like we can just check `batch_size` again instead (and maybe condense these if statements).

Reply:
Removed the rank if statement as it would fail anyway.